### PR TITLE
Refactored `Sender` and `Receiver` to use functional option pattern

### PR DESCRIPTION
**Pull request by mellonnen**
Change that is meant to clean up the API a bit by using [functional option pattern](https://www.sohamkamani.com/golang/options-pattern/) for specifying the options.

For instance, if you want to create a `Sender` with the `UI` option you would do
```go
sender := New(programOpts, WithUi(uiCh))
```
Opposed to 
```go
sender := WithUI(New(programOpts), uiCh)
```
By doing this you can more ergonomically add more options to the constructor like this,
```go
sender := New(programOpts, WithServer(serverOpts), WithPayload(buf, int64(buf.Len())))
```
And adding more options in the future would not change the API.

### CODE REVIEW

**Comment by mellonnen** [comment]
Some Commits from an earlier PR made it in, just a heads up.

File: cmd/portal/main.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ func init() {
 
 	parser.AddCommand("serve",
 		"Serve the Rendezvous server",
-		"The serve command serves the Rendezvous server locally.",
+		"The serve comande serves the Rendezvous server locally.",
```
Comment (ZinoKader) [inline cmd/portal/main.go:64]: comande -> command

File: pkg/receiver/rendezvous_client.go [hunk 2]
```diff
@@ -14,6 +14,8 @@ import (
 	"www.github.com/ZinoKader/portal/tools"
 )
 
+// ConnectToRendezvous establishes the intial communication with Rendezvous server and returns the websocket
```
Comment (ZinoKader) [inline pkg/receiver/rendezvous_client.go:17]: intial -> initial

File: cmd/portal/main.go (hunk 1 above)
Comment (mellonnen) [reply cmd/portal/main.go:64]: LOL

File: pkg/receiver/rendezvous_client.go (hunk 2 above)
Comment (mellonnen) [reply pkg/receiver/rendezvous_client.go:17]: LOL

**Comment by ZinoKader** [comment]
This is much cleaner, nice!

**Pull request merged**